infrastructure/main.tf

```diff
@@ -12,24 +8,34 @@ provider "vault" {
   address = "https://vault.reform.hmcts.net:6200"
 }
 
-data "vault_generic_secret" "cookie-encryption-key" {
-  path = "secret/${var.vault_section}/cmc/cookie/encryption-key"
+data "azurerm_key_vault" "cmc_key_vault" {
+  name = "${local.vaultName}"
+  resource_group_name = "${local.vaultName}"
 }
 
-data "vault_generic_secret" "s2s_secret" {
-  path = "secret/${var.vault_section}/ccidam/service-auth-provider/api/microservice-keys/cmc"
+data "azurerm_key_vault_secret" "cookie_encryption_key" {
+  name = "legal-cookie-encryption-key"
+  vault_uri = "${data.azurerm_key_vault.cmc_key_vault.vault_uri}"
 }
 
-data "vault_generic_secret" "draft_store_secret" {
-  path = "secret/${var.vault_section}/cmc/draft-store/encryption-secrets/legal-frontend"
+data "azurerm_key_vault_secret" "s2s_secret" {
+  name = "cmc-s2s-secret"
+  vault_uri = "${data.azurerm_key_vault.cmc_key_vault.vault_uri}"
 }
 
-data "vault_generic_secret" "postcode-lookup-api-key" {
-  path = "secret/${var.vault_section}/cmc/postcode-lookup/api-key"
+data "azurerm_key_vault_secret" "draft_store_primary" {
+  name = "draft-store-primary"
```

Comment:
I guess you want to prefix it with legal as well.

Reply:
Oops, thanks